src/GitHub.App/Services/TeamExplorerContext.cs

```diff
@@ -80,13 +81,17 @@ public class TeamExplorerContext : ITeamExplorerContext
             gitExt.ActiveRepositoriesChanged += Refresh;
         }
 
-        void StartRefresh() => JoinableTaskFactory.Context.Factory.RunAsync(QueueRefreshAsync).Task.FileAndForget(log);
-        void Refresh() => JoinableTaskFactory.Context.Factory.Run(QueueRefreshAsync);
+        void StartRefresh() => JoinableTaskFactory.RunAsync(QueueRefreshAsync).Task.FileAndForget(log);
+        void Refresh() => JoinableTaskFactory.Run(QueueRefreshAsync);
 
         async Task QueueRefreshAsync()
```

Comment:
Does this method need to be thread-safe? It isn't at the moment. If two callers on different threads call it concurrently, you may find yourself with multiple `RefreshAsync` calls executing simultaneously.